.github/workflows/build-glfw.yml

```diff
@@ -0,0 +1,149 @@
+    - name: Check artifact existence and list folder
+      run: |
+        if [ ! -f share/ext/glfw/glfw-3.4-${{ matrix.platform }}.tar.gz ]; then
+          echo "Artifact not found. Listing contents of share/ext/glfw:"
+          ls -lh share/ext/glfw || echo "Directory missing"
+        fi
+    - name: Archive results
+      uses: actions/upload-artifact@v4
+      with:
+        name: glfw-3.4-${{ matrix.platform }}.tar.gz
+        path: share/ext/glfw/glfw-3.4-${{ matrix.platform }}.tar.gz
+        if-no-files-found: error
+        retention-days: 1
+
+  build-linux-arm:
+    strategy:
+      matrix:
+        platform: [arm64-linux]
+    runs-on: ubuntu-22.04-arm
+    steps:
+        
+    - name: Checkout repo
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      with:
+        ref: '${{env.BUILD_BRANCH}}'
+    - name: Install Python
+      uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c
+      with:
+        python-version: 3.13
+    - name: Install dependencies
+      run: ci/ci.sh install --platform=${{ matrix.platform }}
+    - name: Build GLFW
+      run: |
+        cd share/ext
+        ./build.sh glfw ${{ matrix.platform }}
+    - name: Check artifact existence and list folder
+      run: |
+        if [ ! -f share/ext/glfw/glfw-3.4-${{ matrix.platform }}.tar.gz ]; then
+          echo "Artifact not found. Listing contents of share/ext/glfw:"
+          ls -lh share/ext/glfw || echo "Directory missing"
+        fi
+    - name: Archive results
+      uses: actions/upload-artifact@v4
+      with:
+        name: glfw-3.4-${{ matrix.platform }}.tar.gz
+        path: share/ext/glfw/glfw-3.4-${{ matrix.platform }}.tar.gz
+        if-no-files-found: error
+        retention-days: 1
+
+  build-macos:
+    strategy:
+      matrix:
+        platform: [arm64-macos, x86_64-macos]
+    runs-on: macOS-14
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      with:
+        ref: '${{env.BUILD_BRANCH}}'
+    - name: Install Python
+      uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c
+      with:
+        python-version: 3.13
+    - name: Install dependencies
+      run: ci/ci.sh install --platform=${{ matrix.platform }}
+    - name: Build GLFW
+      run: |
+        cd share/ext
+        ./build.sh glfw ${{ matrix.platform }}
+    - name: Check artifact existence and list folder
+      run: |
+        if [ ! -f share/ext/glfw/glfw-3.4-${{ matrix.platform }}.tar.gz ]; then
+          echo "Artifact not found. Listing contents of share/ext/glfw:"
+          ls -lh share/ext/glfw || echo "Directory missing"
+        fi
+    - name: Archive results
+      uses: actions/upload-artifact@v4
+      with:
+        name: glfw-3.4-${{ matrix.platform }}.tar.gz
+        path: share/ext/glfw/glfw-3.4-${{ matrix.platform }}.tar.gz
+        if-no-files-found: error
+        retention-days: 1
```

Comment:
![CleanShot 2025-04-21 at 11 44 37@2x](https://github.com/user-attachments/assets/7b3f4ec4-f4e9-4f39-99cd-fbbe59f50dda)
all the results available in Summary for one day